core/src/main/java/org/apache/calcite/rel/rules/FilterJoinRule.java

```diff
@@ -110,10 +110,9 @@ protected void perform(RelOptRuleCall call, @Nullable Filter filter,
     if (RelOptUtil.classifyFilters(
         join,
         aboveFilters,
-        joinType,
-        true,
-        !joinType.generatesNullsOnLeft(),
-        !joinType.generatesNullsOnRight(),
+        joinType.canPushIntoFromAbove(),
```

Comment:
this `true` (old code) vs `joinType.canPushIntoFromAbove()` (new code) seems to be a change in the logic. Not sure which one is correct....

Reply:
Note that the old code was checking internally `if (!joinType.isOuterJoin() && pushInto) {`. Thus, even if `true` was passed, the check on the join type was determining whether we were pushing into or not.

Reply:
You're right, good point.